bam_stat.c

```diff
@@ -42,32 +42,43 @@ typedef struct {
     long long n_dup[2];
     long long n_diffchr[2], n_diffhigh[2];
     long long n_secondary[2], n_supp[2];
+    long long n_primary[2], n_pmapped[2], n_pdup[2];
 } bam_flagstat_t;
 
-#define flagstat_loop(s, c) do {                                        \
-        int w = ((c)->flag & BAM_FQCFAIL)? 1 : 0;                       \
-        ++(s)->n_reads[w];                                              \
-        if ((c)->flag & BAM_FSECONDARY ) {                              \
-            ++(s)->n_secondary[w];                                      \
-        } else if ((c)->flag & BAM_FSUPPLEMENTARY ) {                   \
-            ++(s)->n_supp[w];                                           \
-        } else if ((c)->flag & BAM_FPAIRED) {                           \
-            ++(s)->n_pair_all[w];                                       \
-            if (((c)->flag & BAM_FPROPER_PAIR) && !((c)->flag & BAM_FUNMAP) ) ++(s)->n_pair_good[w];    \
-            if ((c)->flag & BAM_FREAD1) ++(s)->n_read1[w];              \
-            if ((c)->flag & BAM_FREAD2) ++(s)->n_read2[w];              \
-            if (((c)->flag & BAM_FMUNMAP) && !((c)->flag & BAM_FUNMAP)) ++(s)->n_sgltn[w];  \
-            if (!((c)->flag & BAM_FUNMAP) && !((c)->flag & BAM_FMUNMAP)) { \
-                ++(s)->n_pair_map[w];                                   \
-                if ((c)->mtid != (c)->tid) {                            \
-                    ++(s)->n_diffchr[w];                                \
-                    if ((c)->qual >= 5) ++(s)->n_diffhigh[w];           \
-                }                                                       \
-            }                                                           \
-        }                                                               \
-        if (!((c)->flag & BAM_FUNMAP)) ++(s)->n_mapped[w];              \
-        if ((c)->flag & BAM_FDUP) ++(s)->n_dup[w];                      \
-    } while (0)
+inline static void flagstat_loop(bam_flagstat_t *s, bam1_core_t *c)
+{
+    do {
```

Comment:
As this is now a static inline function instead of a macro, the `do { ... } while (0)` wrapper is no longer needed and can be removed.